apps/etterlatte-hendelser-pdl/src/main/kotlin/hendelserpdl/leesah/LeesahConsumer.kt

```diff
@@ -0,0 +1,55 @@
+package no.nav.etterlatte.hendelserpdl.leesah
+
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.async
+import kotlinx.coroutines.runBlocking
+import no.nav.etterlatte.hendelserpdl.config.KafkaConsumerConfiguration
+import no.nav.etterlatte.hendelserpdl.config.KafkaEnvironment
+import no.nav.person.pdl.leesah.Personhendelse
+import org.apache.kafka.clients.consumer.KafkaConsumer
+import org.apache.kafka.common.errors.WakeupException
+import org.slf4j.LoggerFactory
+import java.time.Duration
+import java.util.concurrent.atomic.AtomicBoolean
+
+class LeesahConsumer(
+    env: Map<String, String>,
+    private val topic: String,
+    kafkaEnvironment: KafkaConsumerConfiguration = KafkaEnvironment()
+) {
+    val consumer: KafkaConsumer<String, Personhendelse>
+
+    private val logger = LoggerFactory.getLogger(LeesahConsumer::class.java)
+
+    init {
+        consumer = KafkaConsumer<String, Personhendelse>(kafkaEnvironment.generateKafkaConsumerProperties(env))
```

Comment:
Du kan vel like gjerne legge kafkaenvironment på contexten og så slipper du å passe in env til consumeren.

Reply:
27b718a